File: olympics.jl
using CSV, DataFrames, StatsPlots, DataFramesMeta

dat = CSV.read("data/olympicsdata.csv",DataFrame)

datst = @orderby(stack(dat,5:8),:Year)

datst.mpersec = datst.distance ./ datst.value

plots = []
for sex in ("F","M")
    for dist in unique(datst.distance)
        dfsub = filter(row -> row.Sex == sex && row.distance == dist,datst)
        push!(plots, plot(dfsub.Year,dfsub.mpersec,group=dfsub.variable,xlim=(2012,2021),ylim=(7,11),
        linewidth=2,legend=false, title = "$dist m Sex $sex"))
    end
end
pl = plot(plots...,layout=(2,3),size=(600,900))

savefig(pl,"Olympics.png")

File: olympics.R

require(dplyr)
require(tidyr)
require(ggplot2)

dat_orig = read.csv("data/olympicsdata.csv")

dat = pivot_longer(dat_orig, c("place_1", "place_2", "place_3", "place_4"),
names_to = "place", names_prefix = "place_")
dat$"m_per_sec" = dat$distance/dat$value

ggplot(dat, aes(x = Year, y = m_per_sec)) +
geom_point(aes(color = factor(place), shape = factor(Event))) +
geom_line(aes(color = factor(place))) +
facet_grid(Sex ~ distance)
